src/lib/markdown_generators.js

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env node
+
+const yaml = require("js-yaml")
+const markdown = require("markdown-builder")
+const titleCase = require("title-case")
+const TurndownService = require("turndown")
+const tables = require("turndown-plugin-gfm").tables
+const turndownService = new TurndownService()
+turndownService.use(tables)
+const { getCourseImageUrl } = require("./helpers")
+
+const makeTopic = feature => {
+  let topic = ""
+  if (feature["ocw_feature"]) {
+    topic += feature["ocw_feature"]
+  }
+  if (feature["ocw_subfeature"]) {
+    topic += ` - ${feature["ocw_subfeature"]}`
+  }
+  return topic
+}
+
+const generateMarkdownFromJson = courseData => {
+  /*
+    This function takes JSON data parsed from a master.json file and returns markdown data
+    */
+  const markdownData = []
+  let courseHomeMarkdown = generateCourseHomeFrontMatter(courseData)
+  courseHomeMarkdown += generateCourseFeatures(courseData)
+  courseHomeMarkdown += generateCourseCollections(courseData)
+  markdownData.push({
+    name: "_index.md",
+    data: courseHomeMarkdown
+  })
+  let menuIndex = 10
+  courseData["course_pages"].forEach(page => {
+    if (page["text"]) {
+      const pageName = page["short_url"]
+      let courseSectionMarkdown = generateCourseSectionFrontMatter(
+        page["title"],
+        menuIndex
+      )
+      courseSectionMarkdown += generateCourseSectionMarkdown(page, courseData)
+      const sectionData = {
+        name: `sections/${pageName}.md`,
+        data: courseSectionMarkdown
+      }
+      markdownData.push(sectionData)
```

Comment:
We can make this a map and use the index provided as a second parameter for `menuIndex`:

    const coursePagesWithText = courseData["course_pages"].filter(page => page["text"])
    coursePagesWithText.map((page, menuIndex) => {
      const courseSectionMarkdown = generateCourseSectionFrontMatter(
        page["title"],
        (menuIndex + 1) * 10
      )
      courseSectionMarkdown += generateCourseSectionMarkdown(page, courseData)
      return {
        name: `sections/${pageName}.md`,
        data: courseSectionMarkdown
      }
    })